injector/dns_disruption.go

```diff
@@ -206,7 +207,17 @@ func (i *DNSDisruptionInjector) Clean() error {
 	// Remove the net_cls classid for cgroup v1
 	if !i.config.Cgroup.IsCgroupV2() {
 		if err := i.config.Cgroup.Write("net_cls", "net_cls.classid", "0"); err != nil {
-			return fmt.Errorf("unable to write net_cls classid: %w", err)
+			targetExists, existsErr := i.config.TargetContainer.Runtime().MaybeExists(context.Background(), i.config.TargetContainer.ID())
+			if existsErr != nil {
+				i.config.Log.Errorw("we couldn't determine if the target container exists", "cgroupErr", err, "existsErr", existsErr)
+			}
+
+			if !targetExists && os.IsNotExist(err) {
```

Comment:
> edit: ran into a bit of a wall, seems LoadContainer is going to work even if the container is gone, haven't found a place in the containerd lib to tell me if the container is running or not
Should we just assume that if the cgroup file is gone, its ok that we haven't erased our changes to it?

So I assume this would imply to remove all container related changes (`runtime.MaybeExists`) and only keep `os.IsNotExists` and do not return an error in such case, in any other error, return the error wrapped.

This sounds appealing and simpler to me 🤗